Mathlib/RingTheory/Int/Basic.lean

```diff
@@ -131,3 +131,20 @@ theorem eq_pow_of_mul_eq_pow_odd {a b c : ℤ} (hab : IsCoprime a b) {k : ℕ} (
   ⟨eq_pow_of_mul_eq_pow_odd_left hab hk h, eq_pow_of_mul_eq_pow_odd_right hab hk h⟩
 
 end Int
+
+namespace Nat
+
+theorem eq_pow_of_mul_eq_pow_left {a b c k : ℕ} (hab : Coprime a b) (h : a * b = c ^ k) :
```

Comment:
These can go into `Mathlib/Algebra/GCDMonoid/Nat.lean`. (The `Int` ones can't, because `IsCoprime` isn't imported there.)